### DISCUSSION TITLE
Parcel character encoding

### DISCUSSION THREAD

**Discussion by LaraMateo** [parcel-bundler/parcel#6838] [Q&A] [answered]
In Parcel 2 RC we use ES modules, so we need to add the type="module" attribute to the script tags (Otherwise, we get an error message). When we chose a script tag to be a module the code is treated as a JavaScript module, and the processing of the script content is not affected by the charset. My question is: How to specify, to the module, the encoding (e.g "utf-8") so that we can use special characters? 

""" The reason I am asking is because when I try to render a special character like ñ it appears as  \xf1 -> this happens when using Parcel V2 """

`<script type="module" src="~/src/index.js"></script>`

using parcel 2.0.0-rc.0

**Comment by mischnic** [comment]
Can you post your code? This is a bug, similar to https://github.com/parcel-bundler/parcel/issues/6820

**Reply by LaraMateo** [reply]
Hello @mischnic , I have posted my code 😃

**Comment by LaraMateo** [comment, marked as answer]
###  Configuration (.babelrc, package.json, cli command)

Default configuration - no .parcelrc, no .babelrc. Contents of package.json:


```json
{
  "name": "example",
  "main": "index.js",
  "scripts": {
    "start": "parcel",
    "build": "parcel build"
  },
  "devDependencies": {
    "@parcel/packager-raw-url": "^2.0.0-rc.0",
    "@parcel/transformer-image": "^2.0.0-rc.0",
    "@parcel/transformer-webmanifest": "^2.0.0-rc.0",
    "parcel": "^2.0.0-rc.0",
  },
  "dependencies": {
    "@material-ui/core": "^4.12.3",
    "@material-ui/icons": "^4.11.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
```
###  Current Behavior

I am using Material UI TextInput component with a placeholder prop that passes the text "Contraseña"; The letter ñ in the word is not rendering as intended, but as \fx1. The HTLM header have meta UTF-8 and I am using parcel 2.0.0-rc.0 as my bundler.

![KgR2l](https://user-images.githubusercontent.com/73840306/131533649-2db40260-5b52-4dd0-a43e-2e488e165fe8.png)


### Possible Solution

The problem seems to be related to the use of Parcel 2 RC [(Bug report).](https://github.com/parcel-bundler/parcel/issues/6820) It is important to notice that the bad encoding occurs only in attribute values; a temporary solution is to put brackets around the string attribute value, as described in [here](https://stackoverflow.com/questions/68988558/react-project-is-not-displaying-text-as-i-intend).

### Code sample

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="theme-color" content="#000000" />
    <meta
      name="APP"
      content="web site created using react"
    />
    <script src="~/src/index.js" type="module"></script>
    <title>App</title>
  </head>
  <body>
    <div id="root"></div>
  </body>
</html>
```

```js
import React, { StrictMode } from "react";
import ReactDOM from "react-dom";
import TextField from "@material-ui/core/TextField";

const rootElement = document.getElementById("root");

ReactDOM.render(
  <StrictMode>
    <TextField color="secondary" margin="none" placeholder="Contraseña" />
  </StrictMode>,
  rootElement
);
```

**Reply by mischnic** [reply]
> The problem seems to be related to the use of Parcel 2 RC [(Bug report).](https://github.com/parcel-bundler/parcel/issues/6820)

Exactly, you can follow that issue until this is fixed
